Sources/MessagesDisplayDataSource.swift

```diff
@@ -26,7 +26,9 @@ import Foundation
 
 public protocol MessagesDisplayDataSource: class, MessagesDataSource {
 
-    func messageColorFor(_ message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> UIColor
+    func textColorFor(_ message: MessageType) -> UIColor
+    
+    func backgroundColorFor(_ message: MessageType) -> UIColor
```

Comment:
As for the excluded parameters, those are actually pretty important IMO. We should try and pass in as much information as possible if we have it. You can use `indexPath` to set colors based on even or odd numbered sections for example. You can use `messagesCollectionView` to get the total number of sections.